src/test/metric_config_test.cpp

```diff
@@ -0,0 +1,324 @@
+                "shape": {"b": "(1,10) "}
+            }
+        }
+    ],
+    "monitoring":
+        {
+            "metrics":
+            {
+                "enable" : true
+            }
+        }
+})";
+
+TEST_F(MetricsConfigTest, MetricsAllEnabledTest) {
+    SetUpConfig(modelMetricsAllEnabledConfig);
+    std::filesystem::copy("/ovms/src/test/dummy", modelPath, std::filesystem::copy_options::recursive);
+    createConfigFileWithContent(ovmsConfig, configFilePath);
+    ConstructorEnabledModelManager manager;
+
+    auto status = manager.loadConfig(configFilePath);
+    ASSERT_TRUE(status.ok());
+
+    const auto& metricConfig = manager.getMetricConfig();
+    ASSERT_EQ(metricConfig.metricsEnabled, true);
+    ASSERT_EQ(metricConfig.endpointsPath, "/metrics");
+    ASSERT_EQ(metricConfig.requestFailGrpcModelInfer, true);
+    ASSERT_EQ(metricConfig.requestFailRestModelInfer, true);
+    ASSERT_EQ(metricConfig.requestFailRestModelMetadata, true);
+}
+
+static const char* modelMetricsBadEndpoint = R"(
+{
+    "model_config_list": [
+        {
+            "config": {
+                "name": "dummy",
+                "base_path": "/ovms/src/test/dummy",
+                "target_device": "CPU",
+                "model_version_policy": {"latest": {"num_versions":1}},
+                "nireq": 100,
+                "shape": {"b": "(1,10) "}
+            }
+        }
+    ],
+    "monitoring":
+        {
+            "metrics":
+            {
+                "enable" : true,
+                "endpoint_path": "/new..metrics"
+            }
+        }
+})";
+
+TEST_F(MetricsConfigTest, DISABLED_MetricsBadEndpoint) {
+    SetUpConfig(modelMetricsBadEndpoint);
+    std::filesystem::copy("/ovms/src/test/dummy", modelPath, std::filesystem::copy_options::recursive);
+    createConfigFileWithContent(ovmsConfig, configFilePath);
+    ConstructorEnabledModelManager manager;
+
+    auto status = manager.loadConfig(configFilePath);
+    ASSERT_EQ(status, StatusCode::INVALID_METRICS_ENDPOINT) << status.string();
+}
+
+static const char* modelMetricsNegative2 = R"(
+{
+    "model_config_list": [
+        {
+            "config": {
+                "name": "dummy",
+                "base_path": "/ovms/src/test/dummy",
+                "target_device": "CPU",
+                "model_version_policy": {"latest": {"num_versions":1}},
+                "nireq": 100,
+                "shape": {"b": "(1,10) "}
+            }
+        }
+    ],
+    "monitoring":
+        {
+            "metrics":
+            {
+                "enable" : {},
+            }
+        }
+})";
+
+TEST_F(MetricsConfigTest, MetricsNegative2) {
+    SetUpConfig(modelMetricsNegative2);
+    std::filesystem::copy("/ovms/src/test/dummy", modelPath, std::filesystem::copy_options::recursive);
+    createConfigFileWithContent(ovmsConfig, configFilePath);
+    ConstructorEnabledModelManager manager;
+
+    auto status = manager.loadConfig(configFilePath);
+    ASSERT_FALSE(status.ok());
```

Comment:
Minor:
when you test for status it's usually better to test for specific status code to know when you change the behavior. Additionally it's good to have sth like
ASSERT_TRUE(status.ok()) << status.string(); to have quicker debug in case of errors.